game/tests.py

```
@@ -1180,6 +1180,75 @@ def test_get_ai_move_falls_back_to_engine_when_book_empty(self):
+        self.assertFalse(ChessGame._is_valid_opening_book({'': [[6, 4, 4, 4]]}))
+
+    def test_validation_rejects_mixed_type_list(self):
+        """A list with a non-int element must be rejected."""
+        fen = 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq'
+        self.assertFalse(ChessGame._is_valid_opening_book({fen: [[6, 4, 'x', 4]]}))
+
+    def test_load_rejects_invalid_book_file(self):
+        """Corrupt JSON that loads as invalid structure must produce empty dict."""
+        ChessGame._opening_book = None
+        with mock.patch.object(ChessGame, 'OPENING_BOOK_PATH', '/nonexistent/bogus.json'):
+            with mock.patch('builtins.open', mock.mock_open(read_data='[]')):
```

Ruff (0.15.14)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)